public_html/index.php

```
@@ -54,124 +78,131 @@ function safeMarkdown($md, &$outline = null) {
+    }, $md);
 
-    $md = preg_replace('/\*\*(.*?)\*\*/s', '<strong>$1</strong>', $md);
-    $md = preg_replace('/\*(.*?)\*/s', '<em>$1</em>', $md);
-    // Images: escape content and allow only http(s), data:image, or relative URLs
+    // Emphasis and strong
+    $md = preg_replace_callback('/\*\*(.*?)\*\*/s', fn($m) => '<strong>' . htmlspecialchars($m[1], ENT_QUOTES, 'UTF-8') . '</strong>', $md);
+    $md = preg_replace_callback('/\*(.*?)\*/s', fn($m) => '<em>' . htmlspecialchars($m[1], ENT_QUOTES, 'UTF-8') . '</em>', $md);
+    
+    // Images
     $md = preg_replace_callback('/!\[([^\]]*)\]\(([^)]+)\)/', function($m) {
         $alt = htmlspecialchars($m[1], ENT_QUOTES, 'UTF-8');
```

PHPMD (2.15.0)

Avoid unused local variables such as '$alt'. (undefined)

(UnusedLocalVariable)